datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -52,6 +55,79 @@ fn sign_extend_be(b: &[u8]) -> [u8; 16] {
     result
 }
 
+/// Define an adapter iterator for extracting statistics from an iterator of
+/// `ParquetStatistics`
+///
+/// Handles checking if the statistics are present and valid with the correct type
+///
+/// Parameters:
+/// * `$iterator_type` is the name of the iterator type (e.g. `BoolStatsIterator`)
+/// * `$parquet_statistics_type` is the type of the statistics (e.g. `ParquetStatistics::Boolean`)
+/// * `$stat_value_type` is the type of the statistics value (e.g. `bool`)
+macro_rules! make_stats_iterator {
```

Comment:
Here is an idea to create structs like `BoolStatsIterator` which will return an iterator of `Option<ValueStatstics<bool>>` and do the validitiy / type checking